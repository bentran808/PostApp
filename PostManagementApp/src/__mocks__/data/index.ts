const initialUser = {
  id: '',
  avatar: '',
  name: '',
  gender: '',
  email: '',
  phone: '',
  address: '',
  role: ''
};

const mockAdmin = {
  id: '621c4188dece66a40dad2951',
  avatar: 'https://picsum.photos/200',
  name: 'Admin',
  gender: 'male',
  email: 'admin@admin.com',
  phone: '+84 (919) 507231',
  address: '199 Bergen Street, Bowie, Oregon, 7933',
  role: 'admin'
};

const mockUser = {
  id: '621c41883112f1c216433dcf',
  avatar: 'https://picsum.photos/200',
  name: 'Mcintyre Morrison',
  gender: 'male',
  email: 'test1@email.com',
  phone: '+84 (838) 539391',
  address: '313 Menahan Street, Aurora, North Dakota, 9139',
  role: 'user'
};

const mockPost = {
  author: mockAdmin,
  company: 'bb',
  year: 2222,
  type: 'b',
  status: true,
  price: 23333111,
  address: '199 Bergen Street, Bowie, Oregon, 7933',
  title: 'bb update',
  description: 'bb',
  photos: [
    {
      height: 480,
      uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      width: 480,
      fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      type: 'image/jpeg',
      fileSize: 6689
    },
    {
      height: 480,
      uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      width: 480,
      fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      type: 'image/jpeg',
      fileSize: 6689
    }
  ],
  pending: false,
  createdAt: 1647507076963,
  updatedAt: 1647511146749,
  id: 2
};

const mockLikes = [
  {
    author: mockAdmin,
    postId: 2,
    createdAt: 1647509052801,
    updatedAt: 1647509052801,
    id: 1
  }
];

const mockCommentsList = [
  {
    author: mockAdmin,
    postId: 2,
    content: 'test',
    createdAt: 1647509052801,
    updatedAt: 1647509052801,
    id: 1
  },
  {
    author: {
      id: '621c4188e8365653f9326450',
      avatar: 'https://picsum.photos/200',
      name: 'Frost Spence',
      gender: 'male',
      email: 'frostspence@entogrok.com',
      phone: '+84 (859) 508289',
      address: '745 Blake Avenue, Belmont, District Of Columbia, 4099',
      role: 'user'
    },
    postId: 2,
    content: 'test2',
    createdAt: 1647509052801,
    updatedAt: 0,
    id: 2
  }
];

const mockNewPost = {
  author: mockAdmin,
  company: 'qwe',
  year: 2021,
  type: 'qwe',
  status: true,
  price: 10000,
  address: 'qwe',
  title: 'qwe',
  description: 'qwe',
  photos: [],
  pending: true
};

const mockEditedPost = {
  '2': mockPost,
  '3': {
    id: 3,
    ...mockNewPost
  }
};

const mockApprovedPost = {
  '2': mockPost,
  '3': {
    id: 3,
    ...mockNewPost,
    pending: false
  }
};

const mockLike = {
  author: mockAdmin,
  postId: 2,
  createdAt: 1647509052801,
  updatedAt: 1647509052801,
  id: 2
};

const mockNewComment = {
  author: mockAdmin,
  postId: 2,
  content: 'abc',
  createdAt: 1647509052801,
  updatedAt: 1647509052801,
  id: 3
};

const mockPhotos = [
  {
    fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    fileSize: 6689,
    height: 480,
    type: 'image/jpeg',
    uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    width: 480
  },
  {
    fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    fileSize: 6689,
    height: 480,
    type: 'image/jpeg',
    uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    width: 480
  }
];

export {
  initialUser,
  mockAdmin,
  mockUser,
  mockPost,
  mockLikes,
  mockCommentsList,
  mockNewPost,
  mockEditedPost,
  mockApprovedPost,
  mockLike,
  mockNewComment,
  mockPhotos
};
