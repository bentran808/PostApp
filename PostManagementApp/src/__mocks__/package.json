{
    "name": "__mock__",
    "private": true
}