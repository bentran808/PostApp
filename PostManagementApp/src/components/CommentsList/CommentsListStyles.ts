import { StyleSheet } from 'react-native';
import { Colors } from 'react-native-paper';

export const styles = StyleSheet.create({
  commentWrapper: {
    paddingHorizontal: 15,
    paddingVertical: 5
  },
  commentList: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  commentInfo: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  commentAvatar: {
    borderRadius: 25,
    height: 25,
    width: 25
  },
  commentUserName: {
    color: Colors.black,
    fontSize: 14,
    fontWeight: 'bold',
    paddingLeft: 15
  },
  commentTime: {
    fontSize: 12,
    color: '#666',
    marginVertical: 5,
    paddingHorizontal: 15
  },
  commentInput: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 15
  },
  text_sm: {
    fontSize: 12
  }
});
