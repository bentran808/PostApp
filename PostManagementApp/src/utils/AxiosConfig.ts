import axios from 'axios';

export const axiosInstance = axios.create({
    baseURL: 'http://192.168.56.1:3000/',
    headers: {
        'Content-Type': 'application/json'
    }
});
