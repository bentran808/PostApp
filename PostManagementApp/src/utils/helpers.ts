import AsyncStorage from '@react-native-async-storage/async-storage';

const sortDesc = (arr: Post[] = []) => {
  return arr.sort((a, b) => ((a.id || 0) < (b.id || 0) ? 1 : -1));
};

const formatPrice = (price: Number) => {
  return price.toLocaleString('it-IT', {
    style: 'currency',
    currency: 'VND'
  });
};

const getApprovedPosts = (arr: Post[] = []) => arr.filter((item) => !item.pending);

const setAccessToken = async (access_token: string) =>
  await AsyncStorage.setItem('access_token', access_token);

const getAccessToken = async () => await AsyncStorage.getItem('access_token');

const removeAccessToken = async () => await AsyncStorage.removeItem('access_token');

export {
  sortDesc,
  formatPrice,
  getApprovedPosts,
  setAccessToken,
  getAccessToken,
  removeAccessToken
};
