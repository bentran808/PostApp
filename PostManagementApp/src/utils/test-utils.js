import React from 'react';
import { Provider } from 'react-redux';
import { Provider as PaperProvider } from 'react-native-paper';
import { render } from '@testing-library/react-native';
import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import authReducer from 'redux/slices/authSlice';
import postReducer from 'redux/slices/postSlices';

const sagaMiddleware = createSagaMiddleware();
const AllTheProviders = ({ children }) => {
  const store = configureStore({
    reducer: {
      auth: authReducer,
      post: postReducer
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false
      }).concat(sagaMiddleware)
  });
  return (
    <Provider store={store}>
      <PaperProvider>{children}</PaperProvider>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
