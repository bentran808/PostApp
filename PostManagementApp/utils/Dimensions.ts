import {Dimensions} from 'react-native';

const windowHeight = Dimensions.get('window').height;

export {windowHeight};
