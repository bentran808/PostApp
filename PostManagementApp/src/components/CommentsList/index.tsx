import moment from 'moment';
import React from 'react';
import { FlatList, Image, Text, TouchableOpacity, View } from 'react-native';
import { Button, Colors, Menu, TextInput } from 'react-native-paper';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { useAppSelector } from '../../hooks';
import { selectCurrentUser } from '../../redux/slices';
import { colors } from '../../theme/Colors';
import { DefaultAvatar } from '../../theme/Images';
import { windowHeight, windowWidth } from '../../utils/Dimensions';
import { styles } from './CommentsListStyles';

type Props = {
  commentsOfItem: PostComment[];
  visible: string;
  editComment: string;
  editContent: string;
  setVisible: (value: string) => void;
  setEditComment: (value: string) => void;
  onSetEditContent: (value: string) => void;
  onDeleteComment: (id: number) => void;
  onEditComment: (id: number, value: string) => void;
};

const CommentsList = ({
  commentsOfItem,
  visible,
  editComment,
  editContent,
  setVisible,
  setEditComment,
  onSetEditContent,
  onDeleteComment,
  onEditComment
}: Props) => {
  const currentUser = useAppSelector((state) => selectCurrentUser(state));
  const userId = currentUser?.id;
  const isAdmin = currentUser?.role === 'admin';

  return (
    <FlatList
      testID="commentList"
      data={commentsOfItem}
      renderItem={({ item }) => {
        const isAuthorComment = item.author.id === userId || isAdmin;
        return (
          <View style={styles.commentWrapper}>
            <View style={styles.commentList}>
              <View style={styles.commentInfo}>
                <Image
                  defaultSource={DefaultAvatar}
                  source={{
                    uri: item.author.avatar
                  }}
                  style={styles.commentAvatar}
                />
                <Text style={styles.commentUserName}>{item.author.name}</Text>
                <Text style={styles.commentTime}>
                  {moment(new Date(item.updatedAt || 0)).fromNow()}
                </Text>
              </View>
              {isAuthorComment && (
                <Menu
                  visible={visible === `comment-${item.id}`}
                  onDismiss={() => setVisible('')}
                  anchor={
                    <TouchableOpacity
                      onPress={() => setVisible(`comment-${item.id}`)}
                      testID="MenuOfComment"
                    >
                      <Ionicons name="ellipsis-vertical" size={16} color={Colors.black} />
                    </TouchableOpacity>
                  }
                >
                  <Menu.Item
                    onPress={() => {
                      setVisible('');
                      setEditComment(`comment-${item.id}`);
                      onSetEditContent(item.content);
                    }}
                    title="Edit"
                    testID="EditOfComment"
                  />
                  <Menu.Item
                    onPress={() => {
                      setVisible('');
                      onDeleteComment(item.id);
                    }}
                    title="Delete"
                    testID="DeleteOfComment"
                  />
                </Menu>
              )}
            </View>
            {editComment === `comment-${item.id}` ? (
              <View style={styles.commentInput}>
                <TextInput
                  mode="outlined"
                  value={editContent}
                  activeOutlineColor={colors.royalBlue}
                  style={{
                    width: windowWidth / 2,
                    height: windowHeight / 20
                  }}
                  onChangeText={(itemValue) => onSetEditContent(itemValue)}
                  placeholder="Write a comment"
                  right={
                    editContent ? (
                      <TextInput.Icon
                        name="close-circle-outline"
                        size={18}
                        onPress={() => onSetEditContent('')}
                        testID="clearEditComment"
                      />
                    ) : undefined
                  }
                />
                <Button
                  color={colors.royalBlue}
                  labelStyle={styles.text_sm}
                  uppercase={false}
                  disabled={editContent === ''}
                  onPress={() => {
                    onEditComment(item.id, editContent);
                    setEditComment('');
                  }}
                  testID="buttonSave"
                >
                  Save
                </Button>
                <Button
                  color={colors.royalBlue}
                  labelStyle={styles.text_sm}
                  uppercase={false}
                  onPress={() => {
                    setEditComment('');
                    onSetEditContent(item.content);
                  }}
                  testID="buttonCancel"
                >
                  Cancel
                </Button>
              </View>
            ) : (
              <Text>{item.content}</Text>
            )}
          </View>
        );
      }}
    />
  );
};

export default React.memo(CommentsList);
