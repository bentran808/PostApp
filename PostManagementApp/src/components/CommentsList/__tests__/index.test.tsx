import React from 'react';
import * as customHooks from '../../../hooks';
import { fireEvent, render } from '../../../utils/test-utils';
import { mockAdmin, mockCommentsList } from '../../../__mocks__/data';
import CommentsList from '../index';

describe('Comments List Component', () => {
  const spy = jest.spyOn(customHooks, 'useAppSelector');
  spy.mockReturnValue(mockAdmin);

  const props = {
    commentsOfItem: mockCommentsList,
    visible: 'comment-1',
    editComment: 'comment-1',
    editContent: 'test',
    setVisible: jest.fn(),
    setEditComment: jest.fn(),
    onSetEditContent: jest.fn(),
    onDeleteComment: jest.fn(),
    onEditComment: jest.fn()
  };

  test('renders correctly', () => {
    const { getAllByTestId, toJSON } = render(<CommentsList {...props} />);
    const menu = getAllByTestId('MenuOfComment');
    fireEvent.press(menu[0]);
    expect(toJSON()).toMatchSnapshot();
  });

  test('should clear edit comment', () => {
    const { getByTestId } = render(<CommentsList {...props} />);
    const button = getByTestId('clearEditComment');
    fireEvent.press(button);
    expect(props.onSetEditContent).toBeCalledWith('');
  });

  test('should edit comment', () => {
    const { getByPlaceholderText } = render(<CommentsList {...props} />);
    const input = getByPlaceholderText('Write a comment');
    fireEvent.changeText(input, 'asd');
    expect(props.onSetEditContent).toBeCalledWith('asd');
  });

  test('should cancel edit comment', () => {
    const { getByTestId } = render(<CommentsList {...props} />);
    const button = getByTestId('buttonCancel');
    fireEvent.press(button);
    expect(props.onSetEditContent).toBeCalledWith('');
  });

  test('should save edit comment', () => {
    const { getByTestId } = render(<CommentsList {...props} />);
    const button = getByTestId('buttonSave');
    fireEvent.press(button);
    expect(props.onEditComment).toBeCalledWith(1, 'test');
  });
});
