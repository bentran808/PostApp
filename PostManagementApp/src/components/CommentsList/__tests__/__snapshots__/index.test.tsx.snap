// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments List Component renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "author": Object {
            "address": "199 Bergen Street, Bowie, Oregon, 7933",
            "avatar": "https://picsum.photos/200",
            "email": "admin@admin.com",
            "gender": "male",
            "id": "621c4188dece66a40dad2951",
            "name": "Admin",
            "phone": "+84 (919) 507231",
            "role": "admin",
          },
          "content": "test",
          "createdAt": 1647509052801,
          "id": 1,
          "postId": 2,
          "updatedAt": 1647509052801,
        },
        Object {
          "author": Object {
            "address": "745 Blake Avenue, Belmont, District Of Columbia, 4099",
            "avatar": "https://picsum.photos/200",
            "email": "frostspence@entogrok.com",
            "gender": "male",
            "id": "621c4188e8365653f9326450",
            "name": "Frost Spence",
            "phone": "+84 (859) 508289",
            "role": "user",
          },
          "content": "test2",
          "createdAt": 1647509052801,
          "id": 2,
          "postId": 2,
          "updatedAt": 0,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="commentList"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "paddingHorizontal": 15,
              "paddingVertical": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Image
                defaultSource={
                  Object {
                    "testUri": "../../../src/assets/images/default-avatar.jpg",
                  }
                }
                source={
                  Object {
                    "uri": "https://picsum.photos/200",
                  }
                }
                style={
                  Object {
                    "borderRadius": 25,
                    "height": 25,
                    "width": 25,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "paddingLeft": 15,
                  }
                }
              >
                Admin
              </Text>
              <Text
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 12,
                    "marginVertical": 5,
                    "paddingHorizontal": 15,
                  }
                }
              >
                6 days ago
              </Text>
            </View>
            <View
              collapsable={false}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="Menu"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 16,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ïŠ¤
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              style={
                Object {
                  "width": 375,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "borderColor": "rgba(0, 0, 0, 0.54)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                  ]
                }
                testID="text-input-outline"
              />
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "paddingBottom": 0,
                      },
                      Object {
                        "minHeight": 66.7,
                        "paddingTop": 8,
                      },
                    ]
                  }
                >
                  <TextInput
                    editable={true}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Write a comment"
                    placeholderTextColor="rgba(0, 0, 0, 0.54)"
                    selectionColor="#2e64e5"
                    style={
                      Array [
                        Object {
                          "flexGrow": 1,
                          "margin": 0,
                          "paddingHorizontal": 14,
                          "zIndex": 1,
                        },
                        Object {
                          "height": 66.7,
                        },
                        Object {
                          "paddingBottom": 25.35,
                          "paddingTop": 25.35,
                        },
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        Object {
                          "marginRight": 44,
                          "paddingRight": 0,
                        },
                      ]
                    }
                    testID="text-input-outlined"
                    underlineColorAndroid="transparent"
                    value="test"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 24,
                        "justifyContent": "center",
                        "position": "absolute",
                        "width": 24,
                      },
                      Object {
                        "right": 12,
                        "top": 24.35,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    hitSlop={
                      Object {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "margin": 6,
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 13.5,
                            "height": 27,
                            "width": 27,
                          },
                          undefined,
                          Object {
                            "margin": 0,
                          },
                        ],
                      ]
                    }
                    testID="clearEditComment"
                  >
                    <View>
                      <Text
                        accessibilityElementsHidden={true}
                        allowFontScaling={false}
                        importantForAccessibility="no-hide-descendants"
                        pointerEvents="none"
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontSize": 18,
                            },
                            Array [
                              Object {
                                "lineHeight": 18,
                                "transform": Array [
                                  Object {
                                    "scaleX": 1,
                                  },
                                ],
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                            ],
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        ó°…š
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "elevation": 0,
                  "minWidth": 64,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                  ]
                }
                testID="buttonSave"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "letterSpacing": 1,
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          undefined,
                          false,
                          Object {
                            "color": "#2e64e5",
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "fontSize": 12,
                          },
                        ],
                      ]
                    }
                  >
                    Save
                  </Text>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "elevation": 0,
                  "minWidth": 64,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                  ]
                }
                testID="buttonCancel"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "letterSpacing": 1,
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          undefined,
                          false,
                          Object {
                            "color": "#2e64e5",
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "fontSize": 12,
                          },
                        ],
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "paddingHorizontal": 15,
              "paddingVertical": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Image
                defaultSource={
                  Object {
                    "testUri": "../../../src/assets/images/default-avatar.jpg",
                  }
                }
                source={
                  Object {
                    "uri": "https://picsum.photos/200",
                  }
                }
                style={
                  Object {
                    "borderRadius": 25,
                    "height": 25,
                    "width": 25,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "paddingLeft": 15,
                  }
                }
              >
                Frost Spence
              </Text>
              <Text
                style={
                  Object {
                    "color": "#666",
                    "fontSize": 12,
                    "marginVertical": 5,
                    "paddingHorizontal": 15,
                  }
                }
              >
                52 years ago
              </Text>
            </View>
            <View
              collapsable={false}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="Menu"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 16,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ïŠ¤
                </Text>
              </View>
            </View>
          </View>
          <Text>
            test2
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
